### ISSUE TITLE
Recipe for APSIM-Singularity or APSIM-Docker

### ISSUE THREAD

**Issue by BrianCollinss** [APSIMInitiative/APSIMClassic#2062]
I think we have had a couple of posts here discussing the details. I thought it might come handy for people like me (not so familiar with Linux) to have a step-by-step guide to build these containers on their own computers or Virtual Machines.

**Comment by peter-devoil** [comment]
I'll ave a look at this, brian. There are some dated [recipes](https://github.com/APSIMInitiative/APSIMClassic/tree/master/Release/Containers) in the source code if you want to get started.

**Comment by BrianCollinss** [comment]
Cheers Peter. I know of those old ones. I just think (as they are old) they are not working. And some instructions/clarifications would be a great help. For example, to me it is not clear where the old one for APSIM classic singularity downloads APSIM itself from.

**Comment by peter-devoil** [comment]
Classic builds aren't available from apsim.info; but nextgen is. I tried to keep the build system separate from the container construction (there's a lot of source code you dont want in the container)  - perhaps that would be best done all in the one place.

**Comment by BrianCollinss** [comment]
You're right. I think it would be ideal if the source code could be downloaded from Github automatically and compiled and then removed. An even more ideal option is to be able to update the source code inside the singularity so there won't be a need to rebuild it every time there's an update.
